esmvalcore/preprocessor/_regrid.py

```diff
@@ -306,6 +307,61 @@ def _attempt_irregular_regridding(cube, scheme):
     return False
 
 
+def extract_location(cube, location, scheme):
+    """Extract a point using a location name, with interpolation.
+
+    Extracts a single location point from a cube, according
+    to the interpolation scheme ``scheme``.
+
+    The function just retrieves the coordinates of the location and then calls
+    the ``extract_point`` preprocessor.
+
+    It can be use to locate cities and villages, but also mountains or other
+    geographical locations.
+
+    Parameters
+    ----------
+    cube : cube
+        The source cube to extract a point from.
+
+    location : str
+        The reference location. Examples: 'mount everest',
+        'romania','new york, usa'
+
+    scheme : str
+        The interpolation scheme. 'linear' or 'nearest'. No default.
+
+    Returns
+    -------
+    Returns a cube with the extracted point, and with adjusted
+    latitude and longitude coordinates.
+
+    Raises
+    ------
+    ValueError:
+        If location is not supplied as a preprocessor parameter.
+    ValueError:
+        If scheme is not supplied as a preprocessor parameter.
+    ValueError:
+        If given location cannot be found by the geolocator.
+    """
+    if location is None:
+        raise ValueError("Location needs to be specified."
+                         " Examples: 'mount everest', 'romania',"
+                         " 'new york, usa'")
+    if scheme is None:
+        raise ValueError("Interpolation scheme needs to be specified."
+                         " Use either 'linear' or 'nearest'.")
+    geolocator = Nominatim(user_agent='esmvalcore')
+    location = geolocator.geocode(location)
+    if location is None:
+        raise ValueError(f'Requested location {location} can not be found.')
+    logger.debug("Extracting data for %s (%s ºN, %s ºW)", location,
+                 location.latitude, location.longitude)
+
+    return extract_point(cube, location.latitude, location.longitude, scheme)
```

Comment:
```suggestion
    geolocation = geolocator.geocode(location)
    if geolocation is None:
        raise ValueError(f'Requested location {location} can not be found.')
    logger.debug("Extracting data for %s (%s ºN, %s ºW)", geolocation,
                 geolocation.latitude, geolocation.longitude)

    return extract_point(cube, geolocation.latitude, geolocation.longitude, scheme)
```

Reply:
why do you want to rename the variable? if `location` returned by geolocator is None, then it raises, if not it'll go and do the extraction

Reply:
Because of the error message: Just try a wrong location with the current version of this branch, e.g. `location: Gondor`. This will give you the error:

```
ValueError: Requested location None can not be found.
```

But I want

```
ValueError: Requested location Gondor can not be found.
```

Reply:
ah gotcha! The raise error needs to have the actual input location :man_facepalming: Sorry, am slow today